src/SignalR/server/Core/src/HubInvocationContext.cs

```diff
@@ -30,31 +40,51 @@ public HubInvocationContext(HubCallerContext context, Type hubType, string hubMe
         /// <param name="context">Context for the active Hub connection and caller.</param>
         /// <param name="hubMethodName">The name of the Hub method being invoked.</param>
         /// <param name="hubMethodArguments">The arguments provided by the client.</param>
+        [Obsolete("This constructor is obsolete and will be removed in a future version. The recommended alternative is to use the other constructor.")]
         public HubInvocationContext(HubCallerContext context, string hubMethodName, object[] hubMethodArguments)
         {
+#pragma warning disable CS0618 // Type or member is obsolete
             HubMethodName = hubMethodName;
-            HubMethodArguments = hubMethodArguments;
+#pragma warning restore CS0618 // Type or member is obsolete
+            Arguments = hubMethodArguments;
             Context = context;
         }
 
+        internal HubInvocationContext(ObjectMethodExecutor objectMethodExecutor, HubCallerContext context, IServiceProvider serviceProvider, Hub hub, object[] hubMethodArguments)
```

Comment:
```suggestion
        internal HubInvocationContext(ObjectMethodExecutor objectMethodExecutor, HubCallerContext context, IServiceProvider serviceProvider, Hub hub, IReadOnlyList<object> hubMethodArguments)
```

Reply:
Internal constructor was going to stay `object[]`, but public one could be `IReadOnlyList<object>`